modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AbstractJavaCodegen.java

```diff
@@ -1228,6 +1229,8 @@ public void postProcessModelProperty(CodegenModel model, CodegenProperty propert
                 model.imports.add("ArrayList");
             } else if ("set".equals(property.containerType)) {
                 model.imports.add("LinkedHashSet");
+                model.imports.add("JsonDeserialize");
+                property.vendorExtensions.put("x-setter-extra-annotation", "@JsonDeserialize(as = LinkedHashSet.class)");
```

Comment:
FYI. I may rename it to `x-java-setter-extra-annotation` later as this extension is for the Java generators only.